ERCS/erc-XXXX.md

```diff
@@ -77,134 +81,166 @@ Example TWIT manifest at <https://my-crypto-dapp.org/twit.json>:
   const doh = require('dohjs');
   const resolver = new doh.DohResolver('https://1.1.1.1/dns-query');
-  let manifestPath = ""
-  const dnsResp = await resolver.query(domain, 'TXT')
-  for(record of dnsResp.answers){
-    if(!record.data.startsWith("TWIT=")) continue;
 
-    manifestPath = record.data.substring(5) // this should be domain + '/.well-known/twit.json'
-    break
+  let manifestPath = '';
+  const dnsResp = await resolver.query(domain, 'TXT');
+  for (record of dnsResp.answers) {
+    if (!record.data.startsWith('TWIT=')) continue;
+
+    manifestPath = record.data.substring(5); // This should be domain + '/.well-known/twit.json'
+    break;
   }
 
-  // 2. parse the manifest and get they key and algo based on `keyId`
-  const manifestReq = await fetch(manifestPath)
-  const manifest = await manifestReq.json()
-  const keyData = manifest.publicKeys.filter(x => x.id == keyId)
-  if(!keyData){ throw new Error("Could not find the signing key")}
-  const key = keyData.publidKey
-  const alg = keyData.alg
-
-  // 3. verify the signature
-  valid = await crypto.verify(alg, key, signature, requestArg)
-  if(!valid){
-    throw new Error("The data was tampered with")
+  // 3. Parse the manifest and get they key and algo based on `keyId`
+  const manifestReq = await fetch(manifestPath);
+  const manifest = await manifestReq.json();
+  const keyData = manifest.publicKeys.filter((x) => x.id == keyId);
+  if (!keyData) {
+    throw new Error('Could not find the signing key');
   }
-  return await request(requestArg)
+
+  const key = keyData.publicKey;
+  const alg = keyData.alg;
+
+  // 4. Verify the signature
+  const valid = await crypto.verify(alg, key, signature, requestPayload);
+  if (!valid) {
+    throw new Error('The data was tampered with');
+  }
+  return await processRequest(requestPayload);
 }
 ```
 
 ### Wallet UX suggestion
 
 Similarly to the padlock icon for HTTPS, wallets should display a visible indication when TWIT is configured on a domain. This will improve the UX of the end user who will immediately be able to tell
-that interactions between the dApp they are using and the wallet are secure, and this will encourage dApp developper to adopt TWIT, making the overall ecosystem more secure
+that interactions between the dapp they are using and the wallet are secure, and this will encourage dapp developer to adopt TWIT, making the overall ecosystem more secure
 
-When dealing with insecure request, either because the dApp (or an attacker) uses `request` on a domain where TWIT is configured, or because the signature does not match, wallets should warn the user but
+When dealing with insecure request, either because the dapp (or an attacker) uses `request` on a domain where TWIT is configured, or because the signature does not match, wallets should warn the user but
 not block: an eloquently worded warning will increase the transparency enough that end user may opt to cancel the interaction or proceed with the unsafe call.
```

Comment:
AFAIK, browsers have generally stopped showing e.g. padlock icons for HTTPS because it induced a false sense of security. We will probably want to reconsider this section. I can give it a go in a follow-up.

Reply:
they have removed it once https was ubiquitous: instead of showing the padlock when the site is using https, they display a warning when the site is not. Should we follow the same approach ? Until this ERC is widely approved, it feels like there would be benefits in highlighting the dapps that do use vs warning against the one that do not

Reply:
Once this PR is merged, I'll ask our design team for a concrete recommendation here.